src/aead/aead.rs

```diff
@@ -309,6 +310,19 @@ impl Algorithm {
     pub fn nonce_len(&self) -> usize { NONCE_LEN }
 }
 
+#[derive(Eq, PartialEq)]
+#[allow(non_camel_case_types)]
+enum AlgorithmID {
+    AES_128_GCM,
+    AES_256_GCM,
+    CHACHA20_POLY1305,
+}
+
+impl PartialEq for Algorithm {
+    fn eq(&self, other: &Self) -> bool { self.id == other.id }
+}
+
+impl Eq for Algorithm {}
```

Comment:
Did some research and it seems that `derive(Eq)` simply checks for every field, it does not look for existing required impls, like Haskell does.